word_movers_distance/Calculate_WMD_cos_scores.ipynb

```diff
@@ -0,0 +1,3817 @@
+    "orgs['text_no_tags'] = tags_removed_org\n",
+    "rela['text_no_tags'] = tags_removed_rela\n",
+    "\n",
+    "import nltk\n",
+    "nltk.download('stopwords')\n",
+    "from nltk.corpus import stopwords\n",
+    "stop_words = stopwords.words('english')\n",
+    "stop_words.extend(['one', 'two', 'three', 'amp', 'may', 'can', 'new', 'also', 'and'])\n",
+    "\n",
+    "import string\n",
+    "import re\n",
+    "import nltk\n",
+    "\n",
+    "def word_process(tt):\n",
+    "    \"\"\"\n",
+    "    helper function to lower text, remove stop words, numbers, and empty \n",
+    "    \"\"\"\n",
+    "    \n",
+    "    tt = tt.lower()\n",
+    "    \n",
+    "    punc = '''!()-[]{};:'\"\\, <>./?@#$%^&*_~=\\n'''\n",
+    "    # Removing punctuations in string \n",
+    "    # Using loop + punctuation string \n",
+    "\n",
+    "\n",
+    "    for ele in tt:  \n",
+    "        if ele in punc:  \n",
+    "            tt = tt.replace(ele, \" \")  \n",
+    "\n",
+    "    # read tokens\n",
+    "    tokens = tt.split()\n",
+    "    lst = [token.translate(punc).lower() for token in tokens ]\n",
+    "    \n",
+    "    #remove stop words\n",
+    "    filtered = []\n",
+    "    for i in lst:\n",
+    "        if i not in stop_words:\n",
+    "            filtered.append(i)\n",
+    "    \n",
+    "    # removing singular numbers and singular letters\n",
+    "    pattern = '[0-9]'\n",
+    "    filtered = [re.sub(pattern, '', i) for i in filtered] \n",
+    "    new = []\n",
+    "    for inp in filtered:\n",
+    "        new.append(' '.join( [w for w in inp.split() if len(w)>1] ))\n",
+    "        \n",
+    "    # filter out empty strings \n",
+    "    new = [i for i in new if i] \n",
+    "\n",
+    "    dt = [d.split() for d in new]\n",
+    "    \n",
+    "  \n",
+    "    st = ''\n",
+    "    for i in dt:\n",
+    "        st = st + \" \" + i[0]\n",
+    "    return st\n",
+    "    \n",
+    "\n",
+    "cult['processed']  =  cult['text_no_tags'].apply(word_process)\n",
+    "rela['processed']  =  rela['text_no_tags'].apply(word_process)\n",
+    "demog['processed']  =  demog['text_no_tags'].apply(word_process)\n",
+    "orgs['processed']  =  orgs['text_no_tags'].apply(word_process)\n",
+    "\n",
+    "cult_pos = cult[cult['cultural_score']==1]\n",
+    "cult_neg = cult[cult['cultural_score']==0]\n",
+    "\n",
+    "rela_pos = rela[rela['relational_score']==1]\n",
+    "rela_neg = rela[rela['relational_score']==0]\n",
+    "\n",
+    "demog_pos = demog[demog['demographic_score']==1]\n",
+    "demog_neg = demog[demog['demographic_score']==0]\n",
+    "\n",
+    "pos_cult = cult_pos.sample(20, replace = False)\n",
+    "neg_cult = cult_neg.sample(20, replace = False)\n",
+    "\n",
+    "pos_rela = rela_pos.sample(20, replace = False)\n",
+    "neg_rela = rela_neg.sample(20, replace = False)\n",
+    "\n",
+    "pos_demog = demog_pos.sample(20, replace = False)\n",
+    "neg_demog = demog_neg.sample(20, replace = False)\n",
```

Comment:
This seems like a fine way to get a sample of 20 from each labeled set (positive cultural, etc.), but the output somehow ends up with **only 10 positives** for cultural and demographic and 30 negatives (instead of the expected 20/20 split). Do some quality control by checking the number of positives and negatives for each perspective throughout the script. Does it load in a  20/20 split as expected? If so, where does this distribution change?

Reply:
Current implementation seems to sample effectively